From 59f67782b9e382b2d71aeabbd493799b179bdec0 Mon Sep 17 00:00:00 2001
From: Hans Breuer <hans@breuer.org>
Date: Sun, 27 Feb 2011 20:32:47 +0000
Subject: Bug 642764 - use png_jmpbuf() rather than direct member access

Fix from Hanno Boeck to make Dia compile with libpng 1.5
https://bugzilla.gnome.org/show_bug.cgi?id=642764
(cherry picked from commit 8e28086b95578d17386c3be9aea132064be829a0)
---
diff --git a/plug-ins/libart/export_png.c b/plug-ins/libart/export_png.c
index f8bbdc1..33ea6c3 100644
--- a/plug-ins/libart/export_png.c
+++ b/plug-ins/libart/export_png.c
@@ -178,7 +178,7 @@ export_png_ok(GtkButton *button, gpointer userdata)
   }
 
   /* set error handling ... */
-  if (setjmp(png->jmpbuf)) {
+  if (setjmp(png_jmpbuf(png))) {
     fclose(fp);
     png_destroy_write_struct(&png, &info);
     message_error(_("Error occurred while writing PNG"));
--
cgit v0.9
